msteams-platform/m365-apps/build-actions-in-m365.md

```diff
@@ -0,0 +1,165 @@
+this.setState({
+    itemId: itemId
+});
+```
+
+| &nbsp; | Name | Description |
+| --- | --- | --- |
+| &nbsp; | `actionId` | Maps to the Action ID supplied inside the manifest. |
+| &nbsp; | `actionObjects` | Array of corresponding action objects. |
+| &nbsp; | `itemId` | The app receives the ID as the content and uses it to query the Microsoft graph. |
+
+The [ActionInfo](/javascript/api/@microsoft/teams-js/actioninfo) interface helps to enable your app to determine when a user opens a page from an Action and the content that initiated the Action.
+
+```javascript
+app.getContext().then((context: app.Context) => {
+    const actionInfo = context.actionInfo;
+    if (actionInfo && actionInfo.actionId == 'myActionId1') {
+        // Handle specific action
+    }
+});
+```
+
+## Access content through Graph API
+
+After obtaining the `itemId` of the triggering content, you can use the Graph API to read or modify the content, facilitating task completion for your users.
+
+```javascript
+async readActionItem() {
+    try {
+        return await this.graphClient.api(`/users/${this.objectId}/drive/items/${this.itemId}`).get();
+    } catch (error) {
+        console.log("readActionItem", error);
+    }
+}
+```
+
+## Run your app
+
+After you update the app package with the required information, you can run your app in Teams Toolkit.
+
+To run your app in Teams Toolkit follow these steps:
+
+1. Go to Visual Studio Code.
+
+1. Select **File** > **Open Folder...**.
+
+1. Select the folder where your app is created.
+
+1. Select **Select Folder**.
+
+1. From the left pane, select **Run and Debug (Ctrl+Shift+D)**.
+
+1. From the debug dropdown menu, select **Debug in the Microsoft 365 app (Edge) without backend**.
+
+   :::image type="content" source="images/actions-debug.png" alt-text="The screenshot shows actions in debug.":::
+
+1. Select **Yes** if you receive the following security warning:
+
+   :::image type="content" source="~/assets/images/teams-toolkit-v2/first-tab/hw-warning.png" alt-text="Screenshot shows the microsoft warning.":::
+
+A browser window opens with Microsoft 365 home page and your app is available under **Apps**.
+
+You can now preview your Actions in Microsoft 365 app, right-click a file that is supported by your Actions. Actions appear in the context menu, for example **Add todo task**.
+
+:::image type="content" source="images/actions-context-menu.png" alt-text="The screenshot shows the actions in context menu.":::
+
+### Enable Actions in Microsoft Admin Center
+
+In the [Microsoft Admin Center](https://admin.microsoft.com/AdminPortal#/homepage), select **Settings** > **Integrated Apps** > **Upload custom apps** and follow instructions to preinstall your app for entire organization or user groups within your tenant. Ensure that you've enabled the app for [targeted release](/microsoft-365/admin/manage/release-options-in-office-365?view=o365-worldwide&preserve-view=true) to the users in the tenant.
```

Comment:
Please add why we need to do this:

During this preview, we offer the opportunity for your Actions to be delivered to end users through tenant admins.    

To enable this, admins are required to upload the app package containing the Actions via the Microsoft Admin Center (admin.microsoft.com) and acquire this app for targeted release to users in the tenant. Once these steps are completed, users will have access to and can utilize the Actions.  